### PR TITLE
Emit build item when the effective ServiceAccount name is computed

### PR DESCRIPTION
**Pull request by metacosm**
Signed-off-by: Chris Laprun <claprun@redhat.com>

### CODE REVIEW

**Comment by metacosm** [comment]
BTW, if this PR is deemed acceptable, I'd like it to be backported if possible. Let me know if there's anything I can do to make that backport easier.

**Comment by metacosm** [comment]
@yrodiere is that your flaky test failing?

**Comment by yrodiere** [comment]
> @yrodiere is that your flaky test failing?

It is, thanks for the heads-up. But we managed to reproduce it forcefully on CI: https://github.com/quarkusio/quarkus/issues/42479#issuecomment-2349067189
We're looking into it :crossed_fingers:

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VanillaKubernetesProcessor.java [hunk 1]
```diff
@@ -293,7 +296,7 @@ public List<DecoratorBuildItem> createDecorators(ApplicationInfoBuildItem applic
 
         // Handle init Containers and Jobs
         result.addAll(KubernetesCommonHelper.createInitContainerDecorators(KUBERNETES, name, initContainers, result));
-        result.addAll(KubernetesCommonHelper.createInitJobDecorators(KUBERNETES, name, jobs, result));
```
Comment (iocanel) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VanillaKubernetesProcessor.java:296]: Why?

**Comment by iocanel** [comment]
@metacosm: The pull request seems good, but I added a comment about a change that I don't really understand.

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VanillaKubernetesProcessor.java (hunk 1 above)
Comment (metacosm) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VanillaKubernetesProcessor.java:296]: Good catch, not sure how that happened but it definitely didn't happen on purpose.

Comment (metacosm) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VanillaKubernetesProcessor.java]: Restored.

**Review by iocanel — APPROVED** [review]
Looks great! Thanks!

**Comment by metacosm** [comment]
BTW, just to reiterate: I'd like this change to be backported to 3.15 if possible.

**Comment by metacosm** [comment]
Thanks!

**Comment by geoand** [comment]
🙏

**Comment by gsmet** [comment]
We discussed it with @rsvoboda and we think it's too big to be backported to 3.15.

**Pull request merged**